modules/packages/LinearAlgebra.chpl

```diff
@@ -1331,6 +1331,100 @@ proc solve (A: [?Adom] ?eltType, b: [?bdom] eltType) {
   return x;
 }
 
+/* Compute least-squares solution to ``A * x = b``.
+   Compute a vector ``x`` such that the 2-norm ``|b - A x|`` is minimized.
+
+   ``cond`` is the cut-off threshold such that singular values will be
+   considered 0.0. If ``cond < 0.0`` (defaults to ``-1.0``), the treshold will
+   be set to ``max((...A.shape)) * epsilon``, where ``epsilon`` is the machine
+   precision for ``A.eltType``.
+
+   Returns a tuple of ``(x, residues, rank, s)``, where:
+
+  - ``x`` is the the least-squares solution with shape of ``b``
+  - ``residues`` is the square of the 2-norm for each column in ``b - a x`` if ``M > N`` and ``A.rank == n``
+    - ``residues`` is a scalar if ``b`` is 1-D, otherwise it is a 1-element array
+  - ``rank`` is the effective rank of ``A``
+  - ``s`` is the singular values of ``a``
+
+    .. note::
+
+      This procedure depends on the :mod:`LAPACK` module, and will generate a
+      compiler error if ``lapackImpl`` is ``none``.
+*/
+proc leastSquares(A: [] ?t, b: [] t, cond = -1.0) throws
+  where A.rank == 2 && b.rank == 1 && usingLAPACK && isLAPACKType(t)
+{
```

Comment:
Can't remember offhand where we stand on this (1) in general, (2) in LinearAlgebra module, but do we need an overload of this without `where` and give a compilerError there?

Reply:
We're not that consistent for (1) or (2). Do you have a preference as the reviewer? Rather than an overload, I would lean towards checking ranks in the function body and emitting a more helpful error message.

Reply:
I prefer to have an error message coming from the module rather than a resolution error coming from the compiler. Have no preference about how it is done, though.